src/selectors/entities/users.ts

```diff
@@ -371,9 +372,22 @@ export function getFilteredUsersStats(state: GlobalState): any {
     return state.entities.users.filteredStats;
 }
 
-export function searchProfiles(state: GlobalState, term: string, skipCurrent = false, filters?: Filters): Array<UserProfile> {
+export function searchProfilesStartingWithTerm(state: GlobalState, term: string, skipCurrent = false, filters?: Filters): Array<UserProfile> {
```

Comment:
I could make it an if else with keeping the default as term search. That way changes are minimal. I was hoping to remove the startWithTerm entirely at the end of the epic but I probably won't be able to do it

Reply:
Sorry i misunderstood your question i guess, the filter is optional and what kinda of composition do you have in mind?

Reply:
Basically I was thinking that the structure of the selectors `searchProfilesStartingWithTerm` and `searchProfilesMatchingWithTerm` is the same so we could have them as one function but it's not a necessary change.

Reply:
i abstracted a bit more code as it is similar.

> Basically I was thinking that the structure of the selectors `searchProfilesStartingWithTerm` and `searchProfilesMatchingWithTerm` is the same so we could have them as one function but it's not a necessary change.

Based. on the suggestions above what i understood is you want one function instead of these two. For that, we have two options both similar HOF or normal functions they pretty much will be the same as we don't have much composition going on here for other functions to leverage.

HOF pattern

```
searchProfiles(state, term, skipCurrent, filters) => {
  return (newFilterFunc) => {
            const users = getUsers(state);
           const profiles = newFilterFunc(Object.keys(users).map((key) => users[key]), term);
           .....
  }
}
```

and the usage will be in the way

`searchProfiles(state, term, skipCurrent, filters)(filterProfilesMatchingWithTerm)` or 
`searchProfiles(state, term, skipCurrent, filters)(filterProfilesStartingWithTerm)`

I am 1/5 to leave the way it is as it is more readable IMO at the place of usage albeit the extra lines of code. Looks weird with multiple filters at the place of usage.

Let me know if you have strong opinions otherwise, I am gonna let it be the way it is

Reply:
Yes that's what I had in mind but I am 1/5 on it as I already approved, I was just thinking out loud what could 'potentially' be improved but it looks good regardless!